Runtime/Analytics/AmplitudeEventLogger.cs

```diff
@@ -0,0 +1,104 @@
+    {
+        private const string ENDPOINT = "https://analytics-sdk.readyplayer.me/";
+        private const string NO_INTERNET_CONNECTION = "No internet connection.";
+        private const string PRODUCTION = "unity";
+        private const string DEVELOPMENT = "unity-dev";
+
+        private static long sessionId;
+
+        private static bool HasInternetConnection => Application.internetReachability != NetworkReachability.NotReachable;
+
+        public static void SetSessionId(long id)
+        {
+            sessionId = id;
+        }
+
+        public static bool IsSessionIdSet()
+        {
+            return sessionId != 0;
+        }
+
+        public static async void LogEvent(string eventName, Dictionary<string, object> eventProperties = null, Dictionary<string, object> userProperties = null)
+        {
+            var eventData = new Dictionary<string, object>
+            {
+                { AmplitudeKeys.DEVICE_ID, SystemInfo.deviceUniqueIdentifier },
+                { AmplitudeKeys.EVENT_TYPE, eventName },
+                { AmplitudeKeys.PLATFORM, ApplicationData.GetData().UnityPlatform },
+                { AmplitudeKeys.SESSION_ID, sessionId },
+                { AmplitudeKeys.APP_VERSION,ApplicationData.GetData().SDKVersion },
+                { AmplitudeKeys.OPERATING_SYSTEM, SystemInfo.operatingSystem }
+            };
+
+            if (userProperties != null)
+            {
+                eventData.Add(AmplitudeKeys.USER_PROPERTIES, userProperties);
+            }
+
+            if (eventProperties != null)
+            {
+                eventData.Add(AmplitudeKeys.EVENT_PROPERTIES, eventProperties);
+            }
+
+            var payload = new
+            {
+                target = GetAnalyticsTarget(),
+                events = new[]
+                {
+                    eventData
+                }
+            };
+
+            var json = JsonConvert.SerializeObject(payload);
+
+            try
+            {
+                await Dispatch(ENDPOINT, json);
+            }
+            catch (Exception exception)
+            {
+                SDKLogger.Log(nameof(AmplitudeEventLogger), exception);
+            }
+        }
+
+        private static async Task Dispatch(string url, string payload)
+        {
+            if (!HasInternetConnection)
+            {
+                throw new Exception(NO_INTERNET_CONNECTION);
+            }
+
+            var headers = new Dictionary<string, string>
+            {
+                { "Content-Type", "application/json" }
+            };
+
+            var webRequestDispatcher = new WebRequestDispatcher();
+            var response = await webRequestDispatcher.SendRequest<Response>(url, HttpMethod.POST, headers, payload);
+
+            if (!response.IsSuccess)
+            {
+                throw new Exception(response.Error);
+            }
+        }
+        
+        private static string GetAnalyticsTarget()
```

Comment:
An alternative to this function could be to just have a public static string declared up the top, something like this. Not really sure if it is better or worse though :D 

#if RPM_DEVELOPMENT
        public static readonly string TARGET_ENVIRONMENT = "unity";
#else
       public static readonly string TARGET_ENVIRONMENT = "unity-dev";
#endif